libminifi/include/utils/file/FileUtils.h

```diff
@@ -402,6 +403,16 @@ class FileUtils {
 #endif
   }
 
+  static void list_dir_all(std::vector<std::pair<std::string, std::string>> &fileList, const std::string& dir,
```

Comment:
No big reason -- I figured it might allow a user to pass in a vector with entries already in it so list_dir_all could be called multiple times on different directories. I can definitely have it return a vector instead if that's preferred.

Reply:
I would prefer that as this is just a wrapper to make usage more simple and dealing with a return value just makes it more simple.

Reply:
Okay it returns a vector now instead of void.